LaneEcho/LWT-bingo
# 56 Make Responsive & Mobile Friendly!

## Purpose
This gets us to the "Good Enough" stage for mobile responsiveness while we wait for any potential updates to Figma.

## Things to Note
**- Our Material-UI theming has been reorganized (I would like to add this to the README)**
```client/Theme/MainTheme.tsx``` contains the current information for our theme, and another theme object for Dark Mode is found in ```DarkMode.tsx```
It uses the experimental_extendTheme API which extends the default theme of MUI, which I found to avoid repetitive code like declaring the font family in both Dark Mode and Light Mode. We can easily update things like font family and breakpoints in the MainTheme object and they'll still apply to Dark Mode without having to also define them within the DarkMode object. This was a better dev experience when considering responsive style. 
**More details:** 
[MUI Theme](https://mui.com/material-ui/customization/theming/)
[extendTheme](https://mui.com/material-ui/experimental-api/css-theme-variables/customization/#theming)

**- The little boxes on the gameboard that hold our phrases are now called "Square" because I had originally named them "box" which is really confusing when also using MUI Box components for layout**

## To Verify
Run ```npm run dev``` to spin up the game. You can view the bingo game on your mobile device by finding the IP address of the computer and adding the port number that the app is running on (Should be 8080). It should look something like this: ipaddress:8080
